src/libsystemd/sd-device/device-enumerator.c

```diff
@@ -884,9 +884,11 @@ static int parent_crawl_children(sd_device_enumerator *enumerator, const char *p
                 if (!child)
                         return -ENOMEM;
 
-                k = parent_add_child(enumerator, child);
-                if (k < 0)
-                        r = k;
+                if (match_sysname(enumerator, de->d_name)) {
```

Comment:
maybe add a comment why you test this beforehand, instead of letting parent_add_child() test that

Reply:
Done.